OpenSim/Framework/Servers/BaseOpenSimServer.cs

```diff
@@ -311,12 +311,21 @@ public virtual void Startup()
         /// </summary>      
         public virtual void Shutdown()
         {
+            Shutdown(0);
```

Comment:
There is debate as to whether or not to exit with status 0 on a shutdown.  The arguments are:
* 0, and possibly any value from 0 to 50, are very common exit codes.  A library could call exit with that value at any time and defeat the whole purpose of being able to disambiguate a shutdown-and-don't-restart from errors or restart operations that should automatically come back up.  To choose 0 would be to leave ambiguity about whether or not to restart.
* 0 is commonly used to mean "exited successfully" and a shutdown without intent to restart is a successful and proper termination of Halcyon.  To choose some other value and leave 0 for errors is contradictory to standard practice and common expectation.

Either choice that is made will lock in the meaning and cannot be changed in the future as tools will come to rely on the value.